rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbCopyAttributesToEntityViewNodeTest.java

```diff
@@ -0,0 +1,251 @@
+        doAnswer(invocation -> {
+            FutureCallback<Void> callback = invocation.getArgument(4);
+            callback.onSuccess(null);
+            return null;
+        }).when(telemetryServiceMock).saveAndNotify(any(), any(), any(AttributeScope.class), anyList(), any(FutureCallback.class));
+        TbMsg newMsg = TbMsg.newMsg(msg, msg.getQueueName(), msg.getRuleChainId(), msg.getRuleNodeId());
+        doAnswer(invocation -> newMsg).when(ctxMock).newMsg(any(), any(String.class), any(), any(), any(), any());
+
+        node.onMsg(ctxMock, msg);
+
+        verify(entityViewServiceMock).findEntityViewsByTenantIdAndEntityIdAsync(eq(TENANT_ID), eq(DEVICE_ID));
+        verify(telemetryServiceMock).saveAndNotify(eq(TENANT_ID), eq(entityView.getId()), eq(AttributeScope.CLIENT_SCOPE), eq(Collections.emptyList()), any(FutureCallback.class));
+        verify(ctxMock).ack(eq(msg));
+        verify(ctxMock).enqueueForTellNext(eq(newMsg), eq(TbNodeConnectionType.SUCCESS));
+    }
+
+    @Test
+    public void givenAttributesValidityPeriodOutOfStartDateAndEndDate_whenOnMsg_thenDoNothing() {
+        EntityViewId entityViewId = new EntityViewId(UUID.fromString("d117f1a4-24ea-4fdd-b94e-5a472e99d925"));
+        EntityView entityView = new EntityView(entityViewId);
+        entityView.setStartTimeMs(Instant.now().minus(2, ChronoUnit.DAYS).toEpochMilli());
+        entityView.setEndTimeMs(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+
+        mockEntityViewService(entityView);
+
+        TbMsg msg = TbMsg.newMsg(
+                ATTRIBUTES_DELETED, DEVICE_ID, new TbMsgMetaData(Map.of("scope", AttributeScope.CLIENT_SCOPE.name())),
+                "{\"attributes\": [\"attribute1\"]}");
+
+        node.onMsg(ctxMock, msg);
+
+        verify(entityViewServiceMock).findEntityViewsByTenantIdAndEntityIdAsync(eq(TENANT_ID), eq(DEVICE_ID));
+        verify(ctxMock).ack(eq(msg));
+    }
+
+    @Test
+    public void givenEmptyMetadata_whenOnMsg_thenThrowsException() {
+        TbMsg msg = TbMsg.newMsg(
+                ATTRIBUTES_UPDATED, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+
+        node.onMsg(ctxMock, msg);
+
+        ArgumentCaptor<Throwable> throwableCaptor = ArgumentCaptor.forClass(Throwable.class);
+        verify(ctxMock).tellFailure(eq(msg), throwableCaptor.capture());
+        assertThat(throwableCaptor.getValue()).isInstanceOf(IllegalArgumentException.class).hasMessage("Message metadata is empty");
+    }
+
+    @ParameterizedTest
+    @EnumSource(TbMsgType.class)
+    public void givenUnsupportedMsgType_whenOnMsg_thenTellFailure(TbMsgType msgType) {
+        TbMsg msg = TbMsg.newMsg(
+                msgType, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+
+        if (msg.isTypeOneOf(ATTRIBUTES_UPDATED, ATTRIBUTES_DELETED,
+                ACTIVITY_EVENT, INACTIVITY_EVENT, POST_ATTRIBUTES_REQUEST)) {
+            return;
+        }
```

Comment:
We will skip the validation if someone mistakenly removes those types from the list of allowed types. Could you please add a test that will ensure that there will be no tellFailure with  "Unsupported msg type [" + msgType + "]" exception message if the type is supported?